client/client.go

```diff
@@ -64,7 +74,34 @@ func New(id uint32, n, f uint32, stack Stack) (Client, error) {
 	}
 
 	seq := makeSequenceGenerator()
-	return makeRequestHandler(id, seq, stack, buf, f), nil
+	collector := makeReplyCollector(f)
+	return makeRequestHandler(id, seq, stack, collector, buf, f), nil
+}
+
+// New creates an instance of Client given a client ID, total number
+// of replica nodes n, number of tolerated faulty replica nodes f, and
+// a stack of external interfaces.
+func GetClient(id uint32, n, f uint32, stack Stack, collector ReplyCollector) (Client, error) {
```

Comment:
@Naoya-Horiguchi I guess the core idea here was to make `GetClient` a generic constructor for client part of BFT protocol. Then it be specialized with a specific `ReplyConnector` implementation, according to the specific BFT protocol used.

Reply:
@Naoya-Horiguchi That is a good point. Maybe it would be better to supply the collector in the stack type.
@sergefdrv , do you mean `ReplyConnector`? if, so, yes, that is the idea.

Reply:
As far as I can see, `GetClient` is supposed to be a generic version of `New`. If we change the definition of `Stack`, we wouldn't have a default constructor for MinBFT client, i.e. `New` would become the same as `GetClient`. We probably want to keep a default constructor for MinBFT.

Reply:
Yes, that was the intention. To preserve a default constructor for MinBFT and another to supply another protocol.